### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: src/newton_raphson.toit [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+*/
+solve --initial/num=0.0 --goal/num=0.0 --max_iterations=20 --precision=1e9 [--function] [--derivative] [--no_convergence]:
+  x/float := initial.to_float
+  previous := float.NAN
+  max_iterations.repeat: | repeats |
+    top := (function.call x) - goal
+    if top == 0: return x
+    new_x := x - top / (derivative.call x)
+    if new_x == x: return x
+    // A billionth is a suitable precision to aim for with IEEE doubles.
```
Comment (floitsch) [inline src/newton_raphson.toit:80]: Do you want to mention a "billionth" here? It's now a parameter.

File: src/newton_raphson.toit [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      old_diff := (function.call x) - goal
+      new_diff := (function.call new_x) - goal
+      return old_diff.abs < new_diff.abs ? x : new_x
+    previous = x
+    x = new_x
+  no_convergence.call x
+  return x
+
+/**
+See $solve.  This version throws an exception if there is no convergence.
```
Comment (floitsch) [inline src/newton_raphson.toit:92]:
```suggestion
Variant of $(solve --initial --goal --max_iterations --precision [--function] [--derivative] [--no_convergence]).
This version throws an exception if there is no convergence.
```

**Review by floitsch — APPROVED** [review]
LGTM!

File: src/newton_raphson.toit (hunk 1 above)
Comment (erikcorry) [reply src/newton_raphson.toit:80]: I think I'll leave it since it's not a doc comment.

**Pull request merged**